src/Avalonia.Controls/Presenters/TextPresenter.cs

```diff
@@ -627,13 +626,13 @@ public void MoveCaretHorizontal(LogicalDirection direction = LogicalDirection.Fo
                 {
                     var textLine = TextLayout.TextLines[lineIndex];
 
-                    characterHit = textLine.GetNextCaretCharacterHit(new CharacterHit(caretIndex));
+                    characterHit = textLine.GetNextCaretCharacterHit(characterHit);
 
                     caretIndex = characterHit.FirstCharacterIndex + characterHit.TrailingLength;
 
                     if (caretIndex - textLine.TrailingWhitespaceLength == textLine.TextRange.End)
                     {
-                        characterHit = new CharacterHit(caretIndex);
+                        //characterHit = new CharacterHit(caretIndex);
```

Comment:
Is this code commented on purpose? This case does nothing ....

Reply:
Y that is intended just a leftover. I will remove commented code in a bit.